contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,334 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >= 0.6.0 < 0.7.0;
+
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
+
+import {SafeMath} from "./math/SafeMath.sol";
+
+contract RockPaperScissors is Ownable, Pausable {
+
+    using SafeMath for uint256;
+
+    mapping(address => uint256) public balances;
+    mapping(bytes32 => Game) public games;
+
+    address private constant NULL_ADDRESS = address(0);
+    string private constant INVALID_PLAYER_MSG = "Invalid player";
+    string private constant INVALID_MOVE_MSG = "Invalid move";
+    string private constant INVALID_STEP_MSG = "Invalid step";
+    string private constant HASH_MISMATCH_MSG = "Move and secret do not match";
+    string private constant GAME_NOT_EXPIRED_MSG = "Game has not expired";
+    string private constant SECRET_EMPTY_MSG = "Secret is empty";
+
+    uint256 constant FORFEIT_WINDOW = 1 days;
+    
+    enum Steps {
+        UNSTARTED, // 0
+        PLAYER_ONE_MOVED, // 1
+        PLAYER_TWO_MOVE, // 2
+        PLAYER_ONE_REVEAL // 3
```

Comment:
So 2 are past participles and 2 are injunctions? ;)

Reply:
My apologies. 🤦